packages/frontend/src/components/wallet-migration/modals/RotateKeysModal/RotateKeysModal.jsx

```diff
@@ -0,0 +1,300 @@
+                id='migration-modal'
+                onClose={() => {}}
+                modalSize='md'
+            >
+                <ConfirmPassphrase
+                    wordIndex={wordIndex}
+                    userInputValue={userInputValue}
+                    userInputValueWrongWord={userInputValueWrongWord}
+                    finishingSetup={finishingSetupForCurrentAccount}
+                    handleChangeWord={(userInputValue) => {
+                        if (userInputValue.match(/[^a-zA-Z]/)) {
+                            return false;
+                        }
+                        setUserInputValue(userInputValue.trim().toLowerCase());
+                        setUserInputValueWrongWord(false);
+                    }}
+                    handleStartOver={() => {
+                        generateAndSetPhrase();
+                        setConfirmPassphrase(false);
+                        setUserInputValue('');
+                    }}
+                    handleConfirmPassphrase={async () => {
+                        try {
+                            setFinishingSetupForCurrentAccount(true);
+                            if (userInputValue !== currentPassphrase.split(' ')[wordIndex]) {
+                                setUserInputValueWrongWord(true);
+                                return;
+                            }
+                            await handleConfirmPassphrase();
+                            setConfirmPassphrase(false);
+                            setShowConfirmSeedphraseModal(() => false);
+                        } finally {
+                            setFinishingSetupForCurrentAccount(false);
+                        }
+                    }}
+                />
+        
+            </Modal>
+        );
+    }
+    return (
+        <>
+           {showConfirmSeedphraseModal ? (
+               <Modal
+                   modalClass="slim"
+                   id='migration-modal'
+                   onClose={() => {}}
+                   modalSize='lg'
+               >
+                   <SavePassphrase
+                       passPhrase={currentPassphrase}
+                       refreshPhrase={() => {
+                           generateAndSetPhrase();
+                       }}
+                       onClickContinue={() => {
+                           setConfirmPassphrase(true);
+                       }}
+                       onClickCancel = { async () => {
+                           localDispatch({ type: ACTIONS.SET_CURRENT_FAILED_AND_END_PROCESS });
+                           setShowConfirmSeedphraseModal(() => false);
+                       }}
+                   />
+               </Modal>
+           )
+               : (
+                   <Modal
+                       modalClass="slim"
+                       id='migration-modal'
+                       onClose={() => {}}
+                       modalSize='md'
+                       style={{ maxWidth: '431px' }}
```

Comment:
It would be great if we can set the modal size once instead of three times here

Reply:
Yes good idea, I just copied it from a previous modal so probably worth doing a full refactor on this point for all the modals in the flow. I created an issue here -> https://github.com/near/near-wallet/issues/2959